### PR TITLE
fix: default security level preference [LEA-1517] [LEA-1518] [LEA-1524]

### PR DESCRIPTION
**Pull request by edgarkhanzadian**
- display "Secure your wallet" screen
- update tests that include secure and insecure account setup

https://github.com/user-attachments/assets/a6154a05-958d-41a8-9acc-56019cf842d6

### CODE REVIEW

File: apps/mobile/src/store/settings/settings.ts [hunk 1]
```diff
@@ -67,7 +67,7 @@ export const initialState: SettingsState = {
   fiatCurrencyPreference: 'USD',
   networkPreference: WalletDefaultNetworkConfigurationIds.mainnet,
   privacyModePreference: 'visible',
-  securityLevelPreference: 'insecure',
+  securityLevelPreference: 'undefined',
```
Comment (fbwoolf) [inline apps/mobile/src/store/settings/settings.ts:70]: I think this is going to break app auth in settings, why does this need to be `undefined`? Can't we just change the code you initially wrote to handle a default of `insecure`?

Comment (fbwoolf) [reply apps/mobile/src/store/settings/settings.ts:70]: If not, please look at the app auth settings and make sure to cover `undefined` there.

Comment (fbwoolf) [reply apps/mobile/src/store/settings/settings.ts:70]: In the app auth setting, we use a toggle so rn it is just handling 2 states. When a user gets to that point it should be defined though as `secure` or `insecure`? So, maybe ok?

**Review by fbwoolf — APPROVED** [review]

File: apps/mobile/src/store/settings/settings.ts (hunk 1 above)
Comment (fbwoolf) [reply apps/mobile/src/store/settings/settings.ts]: Thanks for addressing, lgtm!

File: apps/mobile/src/hooks/create-wallet.ts [hunk 2]
```diff
@@ -59,7 +59,7 @@ export function useCreateWallet() {
 
   async function navigateAndCreateWallet() {
     switch (securityLevelPreference) {
-      case 'undefined':
+      case 'not-selected':
```
Comment (pete-watters) [inline apps/mobile/src/hooks/create-wallet.ts:62]: This switch should have type `securityLevelPreference: SecurityLevelPreference` right?

**Review by pete-watters — APPROVED** [review]

**Pull request merged**